src/modules/roc_sndio/target_sox/roc_sndio/sox_backend.cpp

```diff
@@ -252,27 +202,59 @@ ISink* SoxBackend::open_sink(core::IAllocator& allocator,
         return NULL;
     }
 
-    if (!sink->open(driver, output)) {
-        return NULL;
+    if (output) {
+        if (driver && is_driver_hidden(driver)) {
+            return NULL;
+        }
+
+        const sox_format_handler_t* handler = sox_write_handler(output, driver, NULL);
+        if (!check_handler_flags(handler, filter_flags)) {
+            return NULL;
+        }
+
+        if (!sink->open(driver, output)) {
+            return NULL;
+        }
+        return sink.release();
```

Comment:
Can we extract this block (is hidden check + hander flags check + open) into a function and reuse here and in a loop? This will make open_source() much shorter.

We can also make it a template function that can work with both sources and sinks and use it in both open_source() and open_sink().